ml/tiktoken-scan

```diff
@@ -0,0 +1,88 @@
+#!/usr/bin/env python3
+import json
+import os
+import tempfile
+import tiktoken
+import sys
+import subprocess
+
+findings = []
+filtered = []
+
+
+def score_cl100k(scan, threshold):
+    encoding = tiktoken.get_encoding("cl100k_base")
+    for finding in scan:
+        if (len(encoding.encode(finding["Secret"])) / len(finding["Secret"])) > float(
+            threshold
+        ):
+            filtered.append(finding)
+    return filtered
+
+
+def score_o200k(scan, threshold):
+    encoding = tiktoken.get_encoding("o200k_base")
+    for finding in scan:
+        if (len(encoding.encode(finding["Secret"])) / len(finding["Secret"])) > float(
+            threshold
+        ):
+            filtered.append(finding)
+    return filtered
+
+
+def score_p50k(scan, threshold):
+    encoding = tiktoken.get_encoding("p50k_base")
+    for finding in scan:
+        if (len(encoding.encode(finding["Secret"])) / len(finding["Secret"])) > float(
+            threshold
+        ):
+            filtered.append(finding)
+    return filtered
+
+
+def score_r50k(scan, threshold):
+    encoding = tiktoken.get_encoding("r50k_base")
+    for finding in scan:
+        if (len(encoding.encode(finding["Secret"])) / len(finding["Secret"])) > float(
+            threshold
+        ):
+            filtered.append(finding)
+    return filtered
```

Comment:
```suggestion
def tiktoken_filter(findings, threshold, encoding_name="cl100k_base"):
    encoding = tiktoken.get_encoding(encoding_name)
    threshold = float(threshold)
    return [
        finding for finding in findings
        if len(encoding.encode(finding["Secret"])) / len(finding["Secret"]) > threshold
    ]
```

Should help cut down on some of the duplication and makes it a bit terser :ok_hand:

Reply:
Then the if/elif block at the bottom could just become

```python
output = tiktoken_filter(stdin_scan, threshold, encoding)
```